viskores/cont/testing/UnitTestCubicHermiteSpline.cxx

```diff
@@ -0,0 +1,189 @@
+{
+  return CheckEvaluation(
+    spline, viskores::cont::make_ArrayHandle(params, viskores::CopyFlag::Off), posAnswer);
+}
+
+//convenience function to save samples for debugging.
+void SaveSamples(viskores::cont::CubicHermiteSpline& spline)
+{
+  std::ofstream fout("pts.txt");
+  fout << "I, X, Y, Z" << std::endl;
+  const auto pts = spline.GetData().ReadPortal();
+  for (viskores::Id i = 0; i < pts.GetNumberOfValues(); ++i)
+    fout << i << ", " << pts.Get(i)[0] << ", " << pts.Get(i)[1] << ", " << pts.Get(i)[2]
+         << std::endl;
+  fout.close();
+
+  fout = std::ofstream("samples.txt");
+  fout << "T, X, Y, Z" << std::endl;
+  viskores::Id n = 1000;
+
+  viskores::FloatDefault t0 = spline.GetParametricRange().Min;
+  viskores::FloatDefault t1 = spline.GetParametricRange().Max;
+  auto dt = (t1 - t0) / static_cast<viskores::FloatDefault>(n - 1);
+  std::vector<viskores::FloatDefault> params;
+  for (viskores::FloatDefault t = t0; t < t1; t += dt)
+    params.push_back(t);
+
+  viskores::cont::Invoker invoke;
+  viskores::cont::ArrayHandle<viskores::Vec3f> posResults;
+  invoke(SplineEvalWorklet{},
+         viskores::cont::make_ArrayHandle(params, viskores::CopyFlag::Off),
+         spline,
+         posResults);
+
+  auto pos = posResults.ReadPortal();
+  for (viskores::Id i = 0; i < pos.GetNumberOfValues(); i++)
+    fout << params[i] << ", " << pos.Get(i)[0] << ", " << pos.Get(i)[1] << ", " << pos.Get(i)[2]
+         << std::endl;
+  fout.close();
+}
+
+void CubicHermiteSplineTest()
+{
+  std::vector<viskores::Vec3f> pts = { { 0, 0, 0 },  { 1, 1, 1 },  { 2, 1, 0 }, { 3, -.5, -1 },
+                                       { 4, -1, 0 }, { 5, -1, 1 }, { 6, 0, 0 } };
+
+  viskores::cont::CubicHermiteSpline spline;
+  spline.SetData(pts);
+  //Evaluation at knots gives the sample pts.
+  CheckEvaluation(spline, spline.GetKnots(), pts);
+  CheckEvaluation(spline, spline.GetKnots(), pts);
+
+  //Evaluation at non-knot values.
+  std::vector<viskores::FloatDefault> params = { 0.21, 0.465, 0.501, 0.99832 };
+  std::vector<viskores::Vec3f> result = { { 1.23261, 1.08861, 0.891725 },
+                                          { 2.68524, -0.0560059, -0.855685 },
+                                          { 2.85574, -0.32766, -0.970523 },
+                                          { 5.99045, -0.00959875, 0.00964856 } };
+  CheckEvaluation(spline, params, result);
+
+  //Explicitly set knots and check.
+  std::vector<viskores::FloatDefault> knots = { 0, 1, 2, 3, 4, 5, 6 };
+  spline = viskores::cont::CubicHermiteSpline();
+  spline.SetData(pts);
+  spline.SetKnots(knots);
+  CheckEvaluation(spline, knots, pts);
+
+  //Evaluation at non-knot values.
+  params = { 0.84, 1.399, 2.838, 4.930, 5.001, 5.993 };
+  result = { { 0.84, 0.896448, 0.952896 },    { 1.399, 1.14382, 0.745119 },
+             { 2.838, -0.297388, -0.951764 }, { 4.93, -1.03141, 0.990543 },
+             { 5.001, -0.999499, 0.999998 },  { 5.993, -0.00702441, 0.00704873 } };
```

Comment:
Change literals to floats to avoid [compiler warnings](https://open.cdash.org/viewBuildError.php?type=1&buildid=10385622).